arm/trafficmanager/usermetricskeys.go

```diff
@@ -0,0 +1,222 @@
+package trafficmanager
+
+// Copyright (c) Microsoft and contributors.  All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// Code generated by Microsoft (R) AutoRest Code Generator.
+// Changes may cause incorrect behavior and will be lost if the code is regenerated.
+
+import (
+	"github.com/Azure/go-autorest/autorest"
+	"github.com/Azure/go-autorest/autorest/azure"
+	"net/http"
+)
+
+// UserMetricsKeysClient is the client for the UserMetricsKeys methods of the Trafficmanager service.
+type UserMetricsKeysClient struct {
+	ManagementClient
+}
+
+// NewUserMetricsKeysClient creates an instance of the UserMetricsKeysClient client.
+func NewUserMetricsKeysClient(subscriptionID string) UserMetricsKeysClient {
+	return NewUserMetricsKeysClientWithBaseURI(DefaultBaseURI, subscriptionID)
+}
+
+// NewUserMetricsKeysClientWithBaseURI creates an instance of the UserMetricsKeysClient client.
+func NewUserMetricsKeysClientWithBaseURI(baseURI string, subscriptionID string) UserMetricsKeysClient {
+	return UserMetricsKeysClient{NewWithBaseURI(baseURI, subscriptionID)}
+}
+
+// CreateOrUpdate create or update a subscription-level key used for Realtime User Metrics collection.
+func (client UserMetricsKeysClient) CreateOrUpdate() (result UserMetricsKeyModel, err error) {
+	req, err := client.CreateOrUpdatePreparer()
+	if err != nil {
+		err = autorest.NewErrorWithError(err, "trafficmanager.UserMetricsKeysClient", "CreateOrUpdate", nil, "Failure preparing request")
+		return
+	}
+
+	resp, err := client.CreateOrUpdateSender(req)
+	if err != nil {
+		result.Response = autorest.Response{Response: resp}
+		err = autorest.NewErrorWithError(err, "trafficmanager.UserMetricsKeysClient", "CreateOrUpdate", resp, "Failure sending request")
+		return
+	}
+
+	result, err = client.CreateOrUpdateResponder(resp)
+	if err != nil {
+		err = autorest.NewErrorWithError(err, "trafficmanager.UserMetricsKeysClient", "CreateOrUpdate", resp, "Failure responding to request")
+	}
+
+	return
+}
+
+// CreateOrUpdatePreparer prepares the CreateOrUpdate request.
+func (client UserMetricsKeysClient) CreateOrUpdatePreparer() (*http.Request, error) {
+	pathParameters := map[string]interface{}{
+		"subscriptionId": autorest.Encode("path", client.SubscriptionID),
+	}
+
+	const APIVersion = "2017-09-01-preview"
```

Comment:
This operation group looks like it is still has the old API version baked into it. Is that correct?

Reply:
It is intentional
https://github.com/Azure/azure-rest-api-specs/pull/1668